compiler/rustc_codegen_llvm/src/asm.rs

```diff
@@ -240,6 +240,8 @@ impl<'ll, 'tcx> AsmBuilderMethods<'tcx> for Builder<'_, 'll, 'tcx> {
                 }
                 InlineAsmArch::RiscV32 | InlineAsmArch::RiscV64 => {
                     constraints.extend_from_slice(&[
+                        "~{fflags}".to_string(),
+                        "~{frm}".to_string(),
```

Comment:
The reference mentions `fflags` but does not mention `frm` or `vxrm`. Should it?

Reply:
I'm not sure because `frm` and `vxrm` are rounding mode registers (the developer explicitly writes).
So, possible resolution might be opposite: removing them from the clobber list (the only reason I included `frm` in this list is because `vxrm` was already there).

Reply:
From the reference:

> - Floating-point exception flags in `fcsr` (`fflags`).
> - Vector extension state (`vtype`, `vl`, `vcsr`).

From RISC-V Unprivileged ISA V20250508 (Section "20.2. Floating-Point Control and Status Register" and "30.3.10. Vector Control and Status (vcsr) Register"):

- `fcsr` is `fflags` (bits 4:0) + `frm` (bits 7:5) + reserved bits.
- `vcsr` is `vxsat` (bit 0) + `vxrm` (bits 2:1) + reserved bits.

IIUC the presence of `vxrm` here matches the reference, but `frm` does not.

Also, changing the floating-point rounding mode in Rust usually causes immediate undefined behavior, so even if we marked `frm` as clobbered, it's unclear whether it can be used properly. (see also https://github.com/rust-lang/stdarch/pull/1478 and docs added in that.)

Reply:
Sounds like this PR shouldn't be merged, then?

Reply:
@Amanieu 
I see, the ABI matter (I missed that).

So, the best solution here will be:
1.  Remove `frm` from the clobber register list (now)
2.  Add vector state sub-registers including `vxrm` (similar to "`fcsr` (`fflags`)") to the reference for clarification (not urgent)

I'll remove `frm` on the next version.